components/omega/doc/design/EOS.md

```diff
@@ -0,0 +1,181 @@
+# Equation of State
+
+## 1 Overview
+The equation of state relates density to the prognostic state variables temperature and salinity. In the case of a non-Boussinesq model, it is also dependent on prognostic pressure. The prognostic temperature, $\theta$, is either potential temperature or Conservative Temperature, depending on the chosen equation of state, and  $S$ is the absolute salinity (linear eos could use practical salinity). Given that the Omega governing equations are  non-Boussinesq, the equation of state class will provide specific volume from the state variables. It will provide methods for computing the specific volume, and its first derivatives.
+
+## 2 Requirements
+
+### 2.1 Requirement: calculate specific volume and density from prognostic temperature and salinity
+Omega will use TEOS-10 as the primary equation of state. The reference implementation of TEOS-10 is contained in the Gibbs Sea Water (GSW) toolbox function. The Omega implementation will be equivalent to the computation in the GSW toolbox function: `gsw_specvol(sa,ct,p)`
+
+### 2.2 Requirement: allow a choice of equations of state
+Omega-1 should include the option for two EOS options. The first is a linear eos, for benchmarking against MPAS-O and to use in simplified cases. The other option should be a well-established version of TEOS-10 (e.g. Roquet 75-term polynomial). 
+
+### 2.3 Requirement: computational efficiency
+As one of the most expensive kernels, the EOS must be as efficient as possible. Developers should be discouraged from calling the EOS too frequently. The fraction of compute time used be the EOS within Omega1 will be compared to fraction used in MPAS-Ocean. 
+
+### 2.4 Requirement: abide by license for the GSW Oceanographic Toolbox
+The GSW toolbox can only be used without modification, so a Omega-specific port of certain functions will be required to support Kokkos/GPU enabled computation. The toolbox without modifications can still be used for verification testing against the Omega implementation.
+
+### 2.5 Requirement: verification
+The Omega implementation should be compared with the test value published in Roquet et al 2015, and the official GSW Oceanographic Toolbox (supported by TEOS-10) to verify correctness.
+
+### 2.6 Requirement: check value range
+The EOS code should include a check on the range of ocean properties to assess use of TEOS-10 (equivalent to the ``ocean funnel'' check), and a truncation of state variables to use the polynomial only in the valid range (similar to the truncation in MPAS-Ocean). 
+
+### 2.7 Requirement: provide adiabatically-displaced density
+The EOS code should be able to be used to calculate density adiabatically displaced to i) the layer above, ii) the surface. This is needed to calculate the Brunt-Vaisala and for mixed-layer parameterizations. This operation should be done as efficiently as possible (e.g. can we reuse parts of the polynomial? this may also be a candidate for Kernel Fusion?).
```

Comment:
maybe just me and I'm fine to not change, but having questions in the design doc is odd.  maybe say (e.g., exploration of reusing parts of the polynomial or kernel fusion)

Reply:
True, this was more of an internal comment than a necessary description. Thanks for the suggestion.